cobbler/items/image.py

```diff
@@ -517,24 +521,20 @@ def display_name(self, display_name: str):
         self._display_name = display_name
 
     @property
-    def supported_boot_loaders(self):
+    def supported_boot_loaders(self) -> List[str]:
         """
         Read only property which represents the subset of settable bootloaders.
 
         :getter: The bootloaders which are available for being set.
         """
-        try:
-            # If we have already loaded the supported boot loaders from the signature, use that data
-            return self._supported_boot_loaders
-        except Exception:
-            # otherwise, refresh from the signatures / defaults
+        if len(self._supported_boot_loaders) == 0:
             self._supported_boot_loaders = signatures.get_supported_distro_boot_loaders(
                 self
             )
-            return self._supported_boot_loaders
+        return self._supported_boot_loaders
 
     @InheritableProperty
-    def boot_loaders(self) -> list:
+    def boot_loaders(self) -> List[str]:
```

Comment:
Also here, shouldn't we make this function to return `self._resolve['boot_loaders']` ?

Reply:
No. This is the same case as with `Distros`. There are no settings that back this field and as such, we need a little bit of extra code to make this work. Images cannot inherit this from a parent item type as there is no parent item type.